testutil/integration/app.go

```diff
@@ -573,15 +646,50 @@ func (app *App) RunMsg(t *testing.T, msg sdk.Msg, option ...RunOption) *codectyp
 	return response
 }
 
+// NextBlocks calls NextBlock numBlocks times
+func (app *App) NextBlocks(t *testing.T, numBlocks int) {
+	t.Helper()
+
+	for i := 0; i < numBlocks; i++ {
+		app.NextBlock(t)
+	}
+}
+
+// emitEvents emits the events from the finalized block to the event manager
+// of the context in the active app.
+func (app *App) emitEvents(t *testing.T, res *abci.ResponseFinalizeBlock) {
+	t.Helper()
+
+	for _, event := range res.Events {
+		var attrs []cosmostypes.Attribute
+		for _, attr := range event.Attributes {
+			attrs = append(attrs, cosmostypes.Attribute{
+				Key:   attr.Key,
+				Value: attr.Value,
+			})
+		}
+		e := cosmostypes.NewEvent(event.Type, attrs...)
+		app.sdkCtx.EventManager().EmitEvent(e)
+
+		// TODO_TECHDEBT: Figure out why the following doesn't work.
+		// See the TODOs in 'testutil/events/filter.go'
+		//
+		// parsedEvent, err := cosmostypes.ParseTypedEvent(abci.Event(event))
+		// require.NoError(t, err)
+		// app.sdkCtx.EventManager().EmitTypedEvent()
```

Comment:
Is this still outstanding or is this resolved by https://github.com/pokt-network/poktroll/pull/610#discussion_r1645906660 ?